### ISSUE TITLE
REST transforms not recognized as such at deploy

### ISSUE THREAD

**Issue by grtjn** [joemfb/mlpm#6]
There is some smart logic that tries to detect if a file in an mlpm package is a REST extension or a REST transform. REST extensions are recognized nicely, but looks like it fails for transforms. This line looks suspicious to me:

https://github.com/joemfb/mlpm/blob/master/lib/package.js#L30

Also, it doesn't seem to recognize XSLT transforms yet, correct? They are difficult to distinguish from XSLT transforms that are not to be used as REST transform. I think it comes down to these lines:

```
  <xsl:param name="context" as="map:map"/>
  <xsl:param name="params" as="map:map"/>
```

**Comment by joemfb** [comment]
That's right, XSLT isn't detected. Are you saying XQuery transforms also aren't detected?

You can explicitly configure asset types. I'll update this with the syntax.

**Comment by grtjn** [comment]
Yes, xqy transforms are not correctly recognized. I think the namespace should end with /transform/ instead of /resource/. Looks like a copy-paste from the line above.

You can easily test this by installing some mlpm package, dropping the rest-api folder from slush-marklogic-node in there, and running mlpm deploy. The filter-docs.xqy gets installed as a module, instead of as a transform..

Thnx, good to know about the config!

**Comment by grtjn** [comment]
PS: what is the difference between `module` and `asset`?

**Comment by joemfb** [comment]
Yeah, I see the copy/paste error in the namespaces. I'll fix that and push a new release.

`module` = XQY/SJS library module. Currently just metadata, doesn't impact behavior in any way. In the future, could be leveraged by the registry for more precise search.

`asset` is totally generic; any file that isn't one of the more specific types.

**Comment by joemfb** [comment]
I just released v0.4.5, which fixes the transform NS bug, and opened a docs bug on the mlpm-registry project.
